metaci/build/tasks.py

```diff
@@ -92,8 +96,39 @@ def run_build(build_id, lock_id=None):
     return build.get_status()
 
 
-def start_build(build, lock_id=None):
-    queue = django_rq.get_queue(build.plan.queue)
+def dispatch_build(build, lock_id: str = None):
+    queue_name = build.plan.queue
+    if queue_name == "long-running":
+        return dispatch_one_off_build(build, lock_id)
+    else:
+        return dispatch_queued_build(build, lock_id)
+
```

Comment:
I'm confused by your approach here.

If you want long-running builds to use check_queued_build in the short queue to make sure that we have an org available (scratch org capacity or a persistent org lock), then how will the lock get released at the end of metaci.build.tasks.run_build? It won't have the lock id.

On the other hand if you want to check to make sure we have an org available when the one-off dyno starts up (like you have in here https://github.com/SFDO-Tooling/MetaCI/pull/2042/files#diff-894872607343abb7ad0807783377d82e2807ec5d8561b9a659fc5e0677280673R48) then I think we may as well call run_one_off_build directly from metaci.build.handlers.queue_build when the Build model is saved, and not go through check_queued_build.

Reply:
Okay, I had some misunderstandings about how locking works and in fact the old run_build task probably didn't handle locking properly I think. (hopefully fixed now)

Still confused about the difference between `metaci.build.management.commands.run_build.lock_org` and `metaci.cumulusci.org.Org.lock()` but that's not relevant.

Reply:
5ec4ac70